blockchain/chain.go

```diff
@@ -1353,6 +1365,7 @@ func (b *BlockChain) reorganizeChain(detachNodes, attachNodes *list.List) error
 			parent = detachBlocks[i+1]
 		}
 		if n.parent.hash != *parent.Hash() {
+			reorgDone()
```

Comment:
No point in doing this before panic, but can be removed anyway with suggested defer above.